src/main/java/de/tum/in/www1/artemis/service/exam/ExamLiveEventsService.java

```diff
@@ -139,6 +139,7 @@ public void createAndSendWorkingTimeUpdateEvent(StudentExam studentExam, int new
      * @param exercise The exam exercise the problem statement was updated for
      * @param message  The message to send
      */
+    @Async
```

Comment:
It won't work, because security context is not propagated to the async thread (tested on ts5, update event is not sent). I suggest we move `userRepository.getUser()` to the calling method and add `User` as a parameter of thr method `createAndSendProblemStatementUpdateEvent`. This way we can also merge the public and protected methods into one to avoid async method calling another async method.